test/gtest/mhp/distributed_dense_matrix3.cpp

```diff
@@ -0,0 +1,245 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+
+using T = int;
+using DM = dr::mhp::distributed_dense_matrix<T>;
+using A = std::allocator<T>;
+using DMA = dr::mhp::distributed_dense_matrix<T, A>;
+using DMI = typename DM::iterator;
+
+template <typename T> class MhpTests3 : public testing::Test {};
+
+TYPED_TEST_SUITE(MhpTests3, AllTypes);
+
+TYPED_TEST(MhpTests3, suite_works_for_3_processes_only) {
+  EXPECT_EQ(dr::mhp::default_comm().size(), 3); // dr-style ignore
+}
+
+TEST(MhpTests3, DM_Index) {
+  assert(dr::mhp::default_comm().size() == 3);
```

Comment:
I don't think it is needed in every tests, since we have a separate test that checks it.
But I don't insist.